src/thermo/EEDFTwoTermApproximation.cpp

```diff
@@ -0,0 +1,699 @@
+/**
+ *  @file EEDFTwoTermApproximation.cpp
+ *  EEDF Two-Term approximation solver.  Implementation file for class
+ *  EEDFTwoTermApproximation.
+ */
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/thermo/EEDFTwoTermApproximation.h"
+#include "cantera/base/ctexceptions.h"
+#include "cantera/numerics/eigen_dense.h"
+#include "cantera/numerics/funcs.h"
+#include "cantera/thermo/PlasmaPhase.h"
+#include "cantera/kinetics/ElectronCollisionPlasmaRate.h"
+
+namespace Cantera
+{
+
+typedef Eigen::SparseMatrix<double> SparseMat;
+
+EEDFTwoTermApproximation::EEDFTwoTermApproximation(PlasmaPhase& s)
+{
+    // store a pointer to s.
+    m_phase = &s;
```

Comment:
Given the need to store a pointer to the phase, it would be more clear that the solver needs to hold onto the phase object if the parameter was a `PlasmaPhase*`.